packages/lib/services/synchronizer/syncInfoUtils.ts

```diff
@@ -96,12 +97,50 @@ export async function fetchSyncInfo(api: FileApi): Promise<SyncInfo> {
 		// If info.json is not present, this might be an old sync target, in
 		// which case we can at least get the version number from version.txt
 		const oldVersion = await api.get('.sync/version.txt');
-		if (oldVersion) output = { version: 1 };
+
+		if (oldVersion) {
+			output = { version: 1 };
+		} else {
+			// Where info.json is missing, but .sync/version.txt is not, the sync target will be set as needing upgrade, and will be upgraded upon restarting the app
+			// If both info.json and .sync/version.txt are missing, it can be assumed that something has gone wrong with the sync target, so do not mark as needing upgrade and raise a failsafe error
+			await performFailsafeValidation(api, checkSyncedItems);
```

Comment:
I found the logic here a bit difficult to understand (see comment for line 140).

**Idea**: Make it clearer that most of the failsafe validation has already been done. Something similar to the following might be more readable:
```suggestion
			// Where info.json is missing, but .sync/version.txt is not, the sync target will be set as needing upgrade, and will be upgraded upon restarting the app
			// If both info.json and .sync/version.txt are missing, it can be assumed that something has gone wrong with the sync target, so do not mark as needing upgrade and raise a failsafe error
			if (!isInitialSync(api.syncTargetId())) {
				failsafeFailure();
			}
```
This suggestion only applies if the `checkSyncItems` parameter can be set to `true` and removed.